Microsoft.Toolkit.Uwp.UI.Animations/CompositionAnimations/Animations/AnimationBase.cs

```diff
@@ -0,0 +1,147 @@
+        /// </summary>
+        public static readonly DependencyProperty ImplicitTargetProperty =
+            DependencyProperty.Register("ImplicitTarget", typeof(string), typeof(AnimationBase), new PropertyMetadata(null, OnAnimationPropertyChanged));
+
+        /// <summary>
+        /// Identifies the <see cref="Delay"/> property
+        /// </summary>
+        public static readonly DependencyProperty DelayProperty =
+            DependencyProperty.Register("Delay", typeof(TimeSpan), typeof(AnimationBase), new PropertyMetadata(TimeSpan.Zero, OnAnimationPropertyChanged));
+
+        /// <summary>
+        /// Raised when a property changes
+        /// </summary>
+        public event EventHandler AnimationChanged;
+
+        /// <summary>
+        /// Gets or sets the duration of the animation
+        /// </summary>
+        public TimeSpan Duration
+        {
+            get { return (TimeSpan)GetValue(DurationProperty); }
+            set { SetValue(DurationProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="KeyFrameCollection"/> of the animations
+        /// </summary>
+        public KeyFrameCollection KeyFrames
+        {
+            get
+            {
+                var collection = (KeyFrameCollection)GetValue(KeyFramesProperty);
+                if (collection == null)
+                {
+                    collection = new KeyFrameCollection();
+                    SetValue(KeyFramesProperty, collection);
+                }
+
+                return collection;
+            }
+
+            set
+            {
+                SetValue(KeyFramesProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the target property to be animated
+        /// </summary>
+        public string Target
+        {
+            get { return (string)GetValue(TargetProperty); }
+            set { SetValue(TargetProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the property that should start the implicit animation
+        /// </summary>
+        public string ImplicitTarget
+        {
+            get { return (string)GetValue(ImplicitTargetProperty); }
+            set { SetValue(ImplicitTargetProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the delay of the animation
+        /// </summary>
+        public TimeSpan Delay
+        {
+            get { return (TimeSpan)GetValue(DelayProperty); }
+            set { SetValue(DelayProperty, value); }
+        }
+
+        private static bool? _isCreatorsUpdateOrAbove;
+
+        /// <summary>
+        /// Gets a value indicating whether Universal API Contract 4 is available on the current device
+        /// where Implicit Show/Hide animations and Translation property on <see cref="Visual"/> are available
+        /// </summary>
+        public static bool IsCreatorsUpdateOrAbove => (bool)(_isCreatorsUpdateOrAbove ??
```

Comment:
This seems like an odd place for this property. Also I think the property should convey whether a feature is supported rather than what version this is.

Reply:
Agree and Disagree. For performance reasons (which I realize is not much), it feels weird having to check each API in a separate property. Agree it doesn't fit in this class, I'll move it out to an internal helper.

Reply:
Moved to internal class